zebra-network/src/peer/connection.rs

```diff
@@ -1056,18 +996,21 @@ where
                     // Always refresh the cache with multi-addr messages.
                     debug!(%msg, "caching unsolicited multi-addr message");
                     self.cached_addrs = addrs.clone();
+                    None
                 } else if addrs.len() == 1 && self.cached_addrs.len() <= 1 {
                     // Only refresh a cached single addr message with another single addr.
                     // (`zcashd` regularly advertises its own address.)
                     debug!(%msg, "caching unsolicited single addr message");
                     self.cached_addrs = addrs.clone();
+                    None
                 } else {
                     debug!(
                         %msg,
                         "ignoring unsolicited single addr message: already cached a multi-addr message"
                     );
+                    unused_msg = Some(msg.clone());
+                    None
```

Comment:
I don't mind making changes here, but I was trying to follow the pattern of the other blocks.

The match statement blocks have 3 different outcomes:
- use the message as an inbound request - Some(Request::_)
- use the message internally (for a ping, cache, or error) - None
- don't use the message - unused_msg = _; None

I think this might be better as a custom enum, let me see what I can do.

Reply:
What do you think of 46018a3?

It's a bit more specific than `Some(Request::_)` and `None`.